### PR TITLE
Ruby 3.3.4: pre-upgrade everything that can apply to Ruby 3.0.5 too

### PR DESCRIPTION
**Pull request by snickell**
https://github.com/code-dot-org/code-dot-org/pull/60329 upgrades us to Ruby 3.3.4. This PR was split out of that branch, and contains all the upgrades that can be pre-applied while we're still on Ruby 3.0.5. As a result, the PR that actually changes the Ruby version is as small as possible.

All details, investigations, and comments on why certain things were upgraded and how can be found in that PR.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ AllCops:
     - 'dashboard/db/schema.rb'
     - 'pegasus/test/test_string.rb' # Parser does not support non-utf8 escape sequences
   DisplayCopNames: true
-  TargetRubyVersion: 2.7
+  TargetRubyVersion: 3.3
```
Comment (snickell) [inline .rubocop.yml:26]: I considered moving this to the "actual upgrade" PR, but `rake lint` does pass set like this, so I guess might as well have it here 🤷

File: SETUP.md [hunk 2]
```diff
@@ -29,8 +29,6 @@ You can do Code.org development using macOS, Ubuntu, or Windows (running Ubuntu
 
 1. `git lfs pull`
 
-1. `gem install bundler -v 2.3.22 && rbenv rehash`
-
```
Comment (snickell) [inline SETUP.md:33]: Arguably this change belongs in the other PR, but this keeps the other PR as small as possible so whatevs.

Comment (Hamms) [reply SETUP.md:33]: Yeah, this seems reasonable to include in whichever PR; certainly I'm not worried about this breaking anything :smile:

File: dashboard/test/integration/honeybadger_test.rb [hunk 3]
```diff
@@ -39,8 +39,9 @@ class HoneybadgerTest < ActionDispatch::IntegrationTest
 
     get raise_error_path
 
-    notice = Honeybadger::Backend::Test.notifications[:notices].first&.as_json
+    # Other tests running in parallel might have logged notices too, but we're only interested in notices about our test controller
```
Comment (Hamms) [inline dashboard/test/integration/honeybadger_test.rb:42]: Thanks! :+1:

**Review by Hamms — APPROVED** [review]
This LGTM, thanks for doing this extraction! I think all the "this PR vs actual upgrade PR" calls you made seem reasonable, and just to be extra cautious I also verified on an adhoc that this subset of changes seems to work fine for Ruby 3.0

Looking forward to the actual upgrade!

**Pull request merged**